spec/factories/registration.rb

```diff
@@ -20,23 +20,44 @@
 
     trait :partnership do
       business_type { "partnership" }
+      people { build_list(:person, 2, :partner) }
     end
 
     trait :sole_trader do
       business_type { "soleTrader" }
     end
 
-    # This is the minimum needed to be able to generate a confirmation email
-    # and its attachments sucessfully
-    trait :emailable do
-      reference { "WEX000999" }
-      submitted_at { Date.today }
-      addresses { [build(:address, :site_address)] }
+    trait :with_manual_site_address do
+      addresses { [build(:address, :site_address, :manual)] }
     end
 
     trait :confirmable do
-      emailable
-      company_no { "123123456" }
+      reference { "WEX000999" }
+      submitted_at { Date.today }
+      limited_company
+      people { [build(:person)] }
+
+      applicant_first_name { "Applicant First Name" }
+      applicant_last_name { "Applicant Last Name" }
+      applicant_phone { "07459745569" }
+      applicant_email { "applicant@test.defra.gov.uk" }
+
+      operator_name { "Operator name" }
+
+      contact_first_name { "Contact First Name" }
+      contact_last_name { "Contact Last Name" }
+      contact_phone { "07459745569" }
+      contact_email { "contact@test.defra.gov.uk" }
+      contact_position { "Manager" }
+
+      addresses { [build(:address, :site_address, :auto), build(:address, :operator_address, :manual), build(:address, :contact_address, :manual)] }
+
+      after(:build) do |registration, _|
```

Comment:
I'm just being nosy here :) What's the `_` argument doing?

Reply:
just convention to say "there is a param there, but I'm not going to use it" :D

Reply:
Ahh OK, I've seen that with a param name afterwards (eg `_foo`) but never just on its own. What is the second param that we are ignoring? The other `after` blocks in our factories only seem to take one param, the object created by the factory.

Reply:
Well, I assume that if you define only one the second just gets lost. I'll remove it is not necessary :) (it is the `evaluator` the object passed there)